### PR TITLE
Added Avazu dataloader

### PR DESCRIPTION
**Pull request by thwang1231**
Avazu dataloader with save/load 1) preprocessed data and 2) fit dictionary:
1. For first 10000 lines (9999 data), achieved cross entropy=0.6239 
2. For 2mil data, achieved cross entropy=0.3718
3. For full dataset, achieved cross entropy = 0.3859

### CODE REVIEW

File: autorecsys/pipeline/preprocessor.py [hunk 1]
```diff
@@ -105,37 +105,207 @@ def preprocessing(self, **kwargs):
...
+        # /home/thwang1231/autorec/examples/datasets/avazu/sampled_train_2mil.txt   # 5/4 loss: 0.3718
+        # /home/thwang1231/autorec/examples/datasets/avazu/train                    # 5/4 loss: 0.3859
+        # Step 1: Set attributes used/set during dataset loading & data preprocessing.
+        self.has_header = True
+        self.column_names = None
+        self.fit_dictionary_path = "./avazu_fit_dictionary.pkl"
+        self.transformed_data_path = "./avazu_transformed_data.pkl"
+        self.categ_filter = 3
+
+        # TODO (for printed message): Notice typo "BinaryCrossentropy" should be "BinaryCrossEntropy"
```
Comment (qingquansong) [inline autorecsys/pipeline/preprocessor.py:120]: tf.keras.metric use BinaryCrossentropy

Comment (thwang1231) [reply autorecsys/pipeline/preprocessor.py:120]: Yes the class is named BinaryCrossentropy in tf.keras.metric. Hence it is not a typo
